lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -91,9 +97,50 @@ private function setInMemcache( $userId, $attributes ) {
 	 * @param Attribute $attribute
 	 */
 	public function setAttribute( $userId, Attribute $attribute ) {
-		$this->setInService( $userId, $attribute );
+		$this->loadAttributes( $userId );
+
+		// If attribute value is null and default exists, set value to default
+		if ( is_null( $attribute->getValue() ) && isset( $this->defaultAttributes[$attribute->getName()] ) ) {
+			$attribute->setValue( $this->defaultAttributes[$attribute->getName()] );
+		}
+
 		$this->setInInstanceCache( $userId, $attribute );
-		$this->setInMemcache( $userId, $this->attributes[$userId] );
+	}
+
+	public function save( $userId ) {
+		$attributes = $this->loadAttributes( $userId );
+
+		// TODO When bulk updates are complete, convert this to a single request
+		$savedAttributes = [];
+		foreach( $attributes as $name => $value ) {
+			if ( $this->attributeShouldBeSaved( $name, $value ) ) {
+				$this->setInService( $userId, new Attribute( $name, $value ) );
+				$savedAttributes[$name] = $value;
+			} elseif ( $this->attributeShouldBeDeleted( $name, $value ) ) {
```

Comment:
Oops, sorry, I think I misunderstood your question. Yes, there is a case. If the attribute doesn't have a default, and has a value of `false` or `null`, then we don't save or delete it. So we do need a second `if`.